dask_cuda/tests/test_dgx.py

```diff
@@ -1,85 +1,122 @@
-async def test_dgx():
-    async with DGX(
-        enable_infiniband=False, enable_nvlink=False, asynchronous=True
-    ) as cluster:
-        async with Client(cluster, asynchronous=True):
-            pass
+def _test_default():
+    with DGX() as cluster:
+        with Client(cluster):
+            res = da.from_array(numpy.arange(10000), chunks=(1000,))
+            res = res.sum().compute()
+            assert res == 49995000
 
 
-@pytest.mark.skipif(
-    "ib0" not in psutil.net_if_addrs(), reason="Infiniband interface ib0 not found"
-)
-@gen_test(timeout=20)
-async def test_dgx_tcp_over_ucx():
-    ucx_env = get_ucx_env(enable_tcp=True)
-    os.environ.update(ucx_env)
+def test_default():
+    p = mp.Process(target=_test_default)
+    p.start()
+    p.join()
+    assert not p.exitcode
 
-    ucp = pytest.importorskip("ucp")
 
-    async with DGX(
-        protocol="ucx", enable_tcp_over_ucx=True, asynchronous=True
-    ) as cluster:
-        async with Client(cluster, asynchronous=True):
-            pass
+def _test_tcp_over_ucx():
+    with DGX(enable_tcp_over_ucx=True) as cluster:
+        with Client(cluster) as client:
+            res = da.from_array(numpy.arange(10000), chunks=(1000,))
+            res = res.sum().compute()
+            assert res == 49995000
 
+            def check_ucx_options():
+                conf = ucp.get_config()
+                assert "TLS" in conf
+                assert "tcp" in conf["TLS"]
+                assert "sockcm" in conf["TLS"]
+                assert "cuda_copy" in conf["TLS"]
+                assert "sockcm" in conf["SOCKADDR_TLS_PRIORITY"]
+                return True
 
-@pytest.mark.skipif(
-    "ib0" not in psutil.net_if_addrs(), reason="Infiniband interface ib0 not found"
-)
-@pytest.mark.parametrize(
-    "params",
-    [
-        {"enable_tcp": True, "enable_infiniband": False, "enable_nvlink": False},
-        {"enable_tcp": True, "enable_infiniband": True, "enable_nvlink": True},
-    ],
-)
-@pytest.mark.asyncio
-async def test_dgx_ucx_infiniband_nvlink(params):
-    ucp = pytest.importorskip("ucp")
+            assert all(client.run(check_ucx_options).values())
 
-    enable_tcp = params["enable_tcp"]
-    enable_infiniband = params["enable_infiniband"]
-    enable_nvlink = params["enable_nvlink"]
 
-    initialize(
-        create_cuda_context=True,
-        enable_tcp_over_ucx=enable_tcp,
-        enable_infiniband=enable_infiniband,
-        enable_nvlink=enable_nvlink,
-    )
+def test_tcp_over_ucx():
+    p = mp.Process(target=_test_tcp_over_ucx)
+    p.start()
+    p.join()
+    assert not p.exitcode
+
+
+def _test_tcp_only():
+    with DGX(protocol="tcp") as cluster:
+        with Client(cluster):
+            res = da.from_array(numpy.arange(10000), chunks=(1000,))
+            res = res.sum().compute()
+            assert res == 49995000
+
+
+def test_tcp_only():
+    p = mp.Process(target=_test_tcp_only)
+    p.start()
+    p.join()
+    assert not p.exitcode
```

Comment:
Do we really need this? It seems to be the same as `test_default`, given that by default TCP will be used.

Reply:
What would yo8u suggest is the default protocol? Right now it is `DGX(protocol="ucx", ...)`. Should it be `tcp` instread?

Reply:
Oh sorry, I have missed that the default has been changed.

I'm not sure what the default should be, it makes sense to have UCX, but I'm not sure if it's not too early to make it default. I think I would keep TCP as default for the time being, @mrocklin do you have a suggestion to untie?

Regardless of what's the default, I see that we need both tests, so let's keep them as they are now.

Reply:
I agree, I think it is best to keep TCP as the default for now. When we are more confident that UCX works out-of-the-box we can change it to the default.

Reply:
> do you have a suggestion to untie?

Maybe this eventually?

```python
try:
    import ucp
except ImportError:
    protocol = "tcp"
else:
    protocol = "ucx"
```

Reply:
Yes

Reply:
Yeah, makes sense once we have stable packages and can make dask-cuda require them.

Reply:
Urm, nevermind. The suggestions is to make protocol _if_ ucp is available. That makes more sense.